### DISCUSSION TITLE
Want to apply template inside a templater script (Nested templates)

### DISCUSSION THREAD

**Discussion by hatimhamid** [SilentVoid13/Templater#1280] [Help]
I have a template file1
```

---
created: <% tp.file.creation_date("YYYY-MM-DD") %>
modified: <% tp.file.last_modified_date("YYYY-MM-DD") %>
---

# hello
```

and a template file2 where I want to include file1 automatically when I apply template file2. File2:
```

---
tags: 
---
/* Add template file1 as well to include created and modified frontmatter automatically. */
```

So now supose I create a file note1 with tenplate file2, it should look something like this:

```
---
created: <date>
modified: <date>
tags: 
---

# hello
```

Basically when I apply template file2 to some other file, file1 should also be applied automatically. (Analogous to Base and Derived template)

Need help in executing this!! Thanks

**Comment by Zachatoo** [comment]
Have you tried using [tp.file.include](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#tpfileincludeinclude_link-string--tfile) for this purpose?

**Reply by hatimhamid** [reply]
@Zachatoo  yes I did...but it just gives this output:

```
---
tags: 
---

---
created: 29-12-2023 20:45
modified: 29-12-2023 22:59
---

# hello
```

Basically it only appends to the existing file instead of adding the frontmatter. I want the frontmatter also to be updated appropriately like this:

```
---
created: 29-12-2023 20:45
modified: 29-12-2023 22:59
tags: 
---

# hello
```

**Reply by Zachatoo** [reply]
It's working as intended for now. Feel free to upvote this feature request if you'd like this functionality to be updated.

#1218

**Reply by hatimhamid** [reply]
true...but when I do it manually using the core plugin, it merges the properties as expected...Do you know how can I invoke the core template plugin in the code. Thanks

**Reply by Zachatoo** [reply]
Yes, the feature request I linked mentions that the core templates plugin works that way.

Templater code (`<% ... %>`) doesn't work in core Templates.

See the comments in that feature request for known workarounds.

**Reply by hatimhamid** [reply]
@Zachatoo sure thanks...coudnt find any workaround as of now! 
Got to find something else as a workaround

**Reply by Zachatoo** [reply]
You didn't see the workaround here?
https://github.com/SilentVoid13/Templater/issues/1218#issuecomment-1806980546

**Reply by hatimhamid** [reply]
I did but its too much work to use processFrontMatter api for every template file and also it doesnt process other markdown elements as expected. I am using other workaround but it creates 3x template files for each file which I want to include but it gets the job done for now. Hoping an official support will be out soon!
